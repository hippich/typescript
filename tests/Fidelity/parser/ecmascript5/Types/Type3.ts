var v: {
   A: B
   <A>(): string
};