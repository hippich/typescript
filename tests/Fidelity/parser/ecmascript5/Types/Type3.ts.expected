{
    "isDeclaration": false,
    "languageVersion": "EcmaScript5",
    "sourceUnit": {
        "kind": "SourceUnit",
        "fullStart": 0,
        "fullEnd": 39,
        "start": 0,
        "end": 39,
        "fullWidth": 39,
        "width": 39,
        "isIncrementallyUnusable": true,
        "moduleElements": [
            {
                "kind": "VariableStatement",
                "fullStart": 0,
                "fullEnd": 39,
                "start": 0,
                "end": 39,
                "fullWidth": 39,
                "width": 39,
                "isIncrementallyUnusable": true,
                "modifiers": [],
                "variableDeclaration": {
                    "kind": "VariableDeclaration",
                    "fullStart": 0,
                    "fullEnd": 38,
                    "start": 0,
                    "end": 38,
                    "fullWidth": 38,
                    "width": 38,
                    "isIncrementallyUnusable": true,
                    "varKeyword": {
                        "kind": "VarKeyword",
                        "fullStart": 0,
                        "fullEnd": 4,
                        "start": 0,
                        "end": 3,
                        "fullWidth": 4,
                        "width": 3,
                        "text": "var",
                        "value": "var",
                        "valueText": "var",
                        "hasTrailingTrivia": true,
                        "trailingTrivia": [
                            {
                                "kind": "WhitespaceTrivia",
                                "fullStart": 3,
                                "fullEnd": 4,
                                "text": " "
                            }
                        ]
                    },
                    "variableDeclarators": [
                        {
                            "kind": "VariableDeclarator",
                            "fullStart": 4,
                            "fullEnd": 38,
                            "start": 4,
                            "end": 38,
                            "fullWidth": 34,
                            "width": 34,
                            "isIncrementallyUnusable": true,
                            "propertyName": {
                                "kind": "IdentifierName",
                                "fullStart": 4,
                                "fullEnd": 5,
                                "start": 4,
                                "end": 5,
                                "fullWidth": 1,
                                "width": 1,
                                "text": "v",
                                "value": "v",
                                "valueText": "v"
                            },
                            "typeAnnotation": {
                                "kind": "TypeAnnotation",
                                "fullStart": 5,
                                "fullEnd": 38,
                                "start": 5,
                                "end": 38,
                                "fullWidth": 33,
                                "width": 33,
                                "isIncrementallyUnusable": true,
                                "colonToken": {
                                    "kind": "ColonToken",
                                    "fullStart": 5,
                                    "fullEnd": 7,
                                    "start": 5,
                                    "end": 6,
                                    "fullWidth": 2,
                                    "width": 1,
                                    "text": ":",
                                    "value": ":",
                                    "valueText": ":",
                                    "hasTrailingTrivia": true,
                                    "trailingTrivia": [
                                        {
                                            "kind": "WhitespaceTrivia",
                                            "fullStart": 6,
                                            "fullEnd": 7,
                                            "text": " "
                                        }
                                    ]
                                },
                                "type": {
                                    "kind": "ObjectType",
                                    "fullStart": 7,
                                    "fullEnd": 38,
                                    "start": 7,
                                    "end": 38,
                                    "fullWidth": 31,
                                    "width": 31,
                                    "isIncrementallyUnusable": true,
                                    "openBraceToken": {
                                        "kind": "OpenBraceToken",
                                        "fullStart": 7,
                                        "fullEnd": 10,
                                        "start": 7,
                                        "end": 8,
                                        "fullWidth": 3,
                                        "width": 1,
                                        "text": "{",
                                        "value": "{",
                                        "valueText": "{",
                                        "hasTrailingTrivia": true,
                                        "hasTrailingNewLine": true,
                                        "trailingTrivia": [
                                            {
                                                "kind": "NewLineTrivia",
                                                "fullStart": 8,
                                                "fullEnd": 10,
                                                "text": "\r\n"
                                            }
                                        ]
                                    },
                                    "typeMembers": [
                                        {
                                            "kind": "PropertySignature",
                                            "fullStart": 10,
                                            "fullEnd": 19,
                                            "start": 13,
                                            "end": 17,
                                            "fullWidth": 9,
                                            "width": 4,
                                            "propertyName": {
                                                "kind": "IdentifierName",
                                                "fullStart": 10,
                                                "fullEnd": 14,
                                                "start": 13,
                                                "end": 14,
                                                "fullWidth": 4,
                                                "width": 1,
                                                "text": "A",
                                                "value": "A",
                                                "valueText": "A",
                                                "hasLeadingTrivia": true,
                                                "leadingTrivia": [
                                                    {
                                                        "kind": "WhitespaceTrivia",
                                                        "fullStart": 10,
                                                        "fullEnd": 13,
                                                        "text": "   "
                                                    }
                                                ]
                                            },
                                            "typeAnnotation": {
                                                "kind": "TypeAnnotation",
                                                "fullStart": 14,
                                                "fullEnd": 19,
                                                "start": 14,
                                                "end": 17,
                                                "fullWidth": 5,
                                                "width": 3,
                                                "colonToken": {
                                                    "kind": "ColonToken",
                                                    "fullStart": 14,
                                                    "fullEnd": 16,
                                                    "start": 14,
                                                    "end": 15,
                                                    "fullWidth": 2,
                                                    "width": 1,
                                                    "text": ":",
                                                    "value": ":",
                                                    "valueText": ":",
                                                    "hasTrailingTrivia": true,
                                                    "trailingTrivia": [
                                                        {
                                                            "kind": "WhitespaceTrivia",
                                                            "fullStart": 15,
                                                            "fullEnd": 16,
                                                            "text": " "
                                                        }
                                                    ]
                                                },
                                                "type": {
                                                    "kind": "IdentifierName",
                                                    "fullStart": 16,
                                                    "fullEnd": 19,
                                                    "start": 16,
                                                    "end": 17,
                                                    "fullWidth": 3,
                                                    "width": 1,
                                                    "text": "B",
                                                    "value": "B",
                                                    "valueText": "B",
                                                    "hasTrailingTrivia": true,
                                                    "hasTrailingNewLine": true,
                                                    "trailingTrivia": [
                                                        {
                                                            "kind": "NewLineTrivia",
                                                            "fullStart": 17,
                                                            "fullEnd": 19,
                                                            "text": "\r\n"
                                                        }
                                                    ]
                                                }
                                            }
                                        },
                                        {
                                            "kind": "SemicolonToken",
                                            "fullStart": 19,
                                            "fullEnd": 19,
                                            "start": 19,
                                            "end": 19,
                                            "fullWidth": 0,
                                            "width": 0,
                                            "text": ""
                                        },
                                        {
                                            "kind": "CallSignature",
                                            "fullStart": 19,
                                            "fullEnd": 37,
                                            "start": 22,
                                            "end": 35,
                                            "fullWidth": 18,
                                            "width": 13,
                                            "typeParameterList": {
                                                "kind": "TypeParameterList",
                                                "fullStart": 19,
                                                "fullEnd": 25,
                                                "start": 22,
                                                "end": 25,
                                                "fullWidth": 6,
                                                "width": 3,
                                                "lessThanToken": {
                                                    "kind": "LessThanToken",
                                                    "fullStart": 19,
                                                    "fullEnd": 23,
                                                    "start": 22,
                                                    "end": 23,
                                                    "fullWidth": 4,
                                                    "width": 1,
                                                    "text": "<",
                                                    "value": "<",
                                                    "valueText": "<",
                                                    "hasLeadingTrivia": true,
                                                    "leadingTrivia": [
                                                        {
                                                            "kind": "WhitespaceTrivia",
                                                            "fullStart": 19,
                                                            "fullEnd": 22,
                                                            "text": "   "
                                                        }
                                                    ]
                                                },
                                                "typeParameters": [
                                                    {
                                                        "kind": "TypeParameter",
                                                        "fullStart": 23,
                                                        "fullEnd": 24,
                                                        "start": 23,
                                                        "end": 24,
                                                        "fullWidth": 1,
                                                        "width": 1,
                                                        "identifier": {
                                                            "kind": "IdentifierName",
                                                            "fullStart": 23,
                                                            "fullEnd": 24,
                                                            "start": 23,
                                                            "end": 24,
                                                            "fullWidth": 1,
                                                            "width": 1,
                                                            "text": "A",
                                                            "value": "A",
                                                            "valueText": "A"
                                                        }
                                                    }
                                                ],
                                                "greaterThanToken": {
                                                    "kind": "GreaterThanToken",
                                                    "fullStart": 24,
                                                    "fullEnd": 25,
                                                    "start": 24,
                                                    "end": 25,
                                                    "fullWidth": 1,
                                                    "width": 1,
                                                    "text": ">",
                                                    "value": ">",
                                                    "valueText": ">"
                                                }
                                            },
                                            "parameterList": {
                                                "kind": "ParameterList",
                                                "fullStart": 25,
                                                "fullEnd": 27,
                                                "start": 25,
                                                "end": 27,
                                                "fullWidth": 2,
                                                "width": 2,
                                                "openParenToken": {
                                                    "kind": "OpenParenToken",
                                                    "fullStart": 25,
                                                    "fullEnd": 26,
                                                    "start": 25,
                                                    "end": 26,
                                                    "fullWidth": 1,
                                                    "width": 1,
                                                    "text": "(",
                                                    "value": "(",
                                                    "valueText": "("
                                                },
                                                "parameters": [],
                                                "closeParenToken": {
                                                    "kind": "CloseParenToken",
                                                    "fullStart": 26,
                                                    "fullEnd": 27,
                                                    "start": 26,
                                                    "end": 27,
                                                    "fullWidth": 1,
                                                    "width": 1,
                                                    "text": ")",
                                                    "value": ")",
                                                    "valueText": ")"
                                                }
                                            },
                                            "typeAnnotation": {
                                                "kind": "TypeAnnotation",
                                                "fullStart": 27,
                                                "fullEnd": 37,
                                                "start": 27,
                                                "end": 35,
                                                "fullWidth": 10,
                                                "width": 8,
                                                "colonToken": {
                                                    "kind": "ColonToken",
                                                    "fullStart": 27,
                                                    "fullEnd": 29,
                                                    "start": 27,
                                                    "end": 28,
                                                    "fullWidth": 2,
                                                    "width": 1,
                                                    "text": ":",
                                                    "value": ":",
                                                    "valueText": ":",
                                                    "hasTrailingTrivia": true,
                                                    "trailingTrivia": [
                                                        {
                                                            "kind": "WhitespaceTrivia",
                                                            "fullStart": 28,
                                                            "fullEnd": 29,
                                                            "text": " "
                                                        }
                                                    ]
                                                },
                                                "type": {
                                                    "kind": "StringKeyword",
                                                    "fullStart": 29,
                                                    "fullEnd": 37,
                                                    "start": 29,
                                                    "end": 35,
                                                    "fullWidth": 8,
                                                    "width": 6,
                                                    "text": "string",
                                                    "value": "string",
                                                    "valueText": "string",
                                                    "hasTrailingTrivia": true,
                                                    "hasTrailingNewLine": true,
                                                    "trailingTrivia": [
                                                        {
                                                            "kind": "NewLineTrivia",
                                                            "fullStart": 35,
                                                            "fullEnd": 37,
                                                            "text": "\r\n"
                                                        }
                                                    ]
                                                }
                                            }
                                        }
                                    ],
                                    "closeBraceToken": {
                                        "kind": "CloseBraceToken",
                                        "fullStart": 37,
                                        "fullEnd": 38,
                                        "start": 37,
                                        "end": 38,
                                        "fullWidth": 1,
                                        "width": 1,
                                        "text": "}",
                                        "value": "}",
                                        "valueText": "}"
                                    }
                                }
                            }
                        }
                    ]
                },
                "semicolonToken": {
                    "kind": "SemicolonToken",
                    "fullStart": 38,
                    "fullEnd": 39,
                    "start": 38,
                    "end": 39,
                    "fullWidth": 1,
                    "width": 1,
                    "text": ";",
                    "value": ";",
                    "valueText": ";"
                }
            }
        ],
        "endOfFileToken": {
            "kind": "EndOfFileToken",
            "fullStart": 39,
            "fullEnd": 39,
            "start": 39,
            "end": 39,
            "fullWidth": 0,
            "width": 0,
            "text": ""
        }
    },
    "lineMap": {
        "lineStarts": [
            0,
            10,
            19,
            37
        ],
        "length": 39
    }
}